com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/MaterialBuilder.java

```diff
@@ -254,6 +265,18 @@ public void build(Task<Void> task) throws CompileExceptionError, IOException {
         BuilderUtil.checkResource(this.project, res, "fragment program", fragmentBuildContext.buildPath);
         materialBuilder.setFragmentProgram(BuilderUtil.replaceExt(fragmentBuildContext.projectPath, ".fp", ".fpc"));
 
+        for (int i=0; i < materialBuilder.getSamplersCount(); i++) {
+            MaterialDesc.Sampler materialSampler = materialBuilder.getSamplers(i);
+            String texture = materialSampler.getTexture();
+            if (texture.isEmpty())
+                continue;
```

Comment:
Isn't this too early to continue? Currently it looks like the name hash is only set if the sampler has a texture assigned.

Reply:
For correctness, I agree and I'll fix that.
Only usage currently is with a texture, that's why it wasn't noticed.

Reply:
I think you need to call `materialBuilder.setSamplers(i, samplerBuilder.build())` to commit the updated `Sampler`? 

If you `continue` here, that won't happen unless a texture is assigned.